pkg/generate/app/cmd/newapp.go

```diff
@@ -443,6 +450,16 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 		}
 		pipelines = append(pipelines, common...)
 	}
+
+	if len(c.BuildArgs) > 0 {
+		if numDockerBuilds == 0 {
+			return nil, fmt.Errorf("Cannot use build arguments without a Docker build")
```

Comment:
is this the error that gets reported to the user?  "build arguments" is probably too vague (all arguments to "oc new-build" are build arguments).  Specify --build-args (or whatever the flag name is) instead.